src/Compilers/CSharp/Portable/Binder/BinderFactory.cs

```diff
@@ -129,10 +128,10 @@ internal Binder GetBinder(SyntaxNode node, int position, CSharpSyntaxNode member
                 container.AssertMemberExposure(memberOpt);
             }
 #endif
-            BinderFactoryVisitor visitor = _binderFactoryVisitorPool.Allocate();
-            visitor.Initialize(position, memberDeclarationOpt, memberOpt);
+            BinderFactoryVisitor visitor = s_binderFactoryVisitorPool.Allocate();
+            visitor.Initialize(factory: this, position, memberDeclarationOpt, memberOpt);
             Binder result = visitor.Visit(node);
-            _binderFactoryVisitorPool.Free(visitor);
+            s_binderFactoryVisitorPool.Free(visitor);
```

Comment:
Would it be useful to clear the instance (remove the `this` reference) before returning it to the pool?

Reply:
Sure, would it be preferable to clear out the BinderFactoryVisitor instance completely (CSharpSyntaxNode/Symbol/BinderFactory references), or are you just worried about the BinderFactory reference being kept about?

Reply:
I'd clear all that's passed in `Initialize`.